dataactbroker/handlers/fileHandler.py

```diff
@@ -890,6 +890,118 @@ def generateFile(self):
         # Return same response as check generation route
         return self.checkGeneration(submission_id, file_type)
 
+    def generate_detached_file(self):
+        """ Start a file generation job for the specified file type """
+        _debug_logger.debug('Starting D file generation')
+        request_dict = RequestDictionary(self.request)
+
+        if not(request_dict.exists("file_type") and request_dict.exists("cgac_code")):
+            raise ResponseException("Generate detached file route requires file_type and cgac_code", StatusCode.CLIENT_ERROR)
+
+        # Get file_type (D1 | D2) and cgac_code from the request dict
+        file_type = request_dict.getValue("file_type")
+        cgac_code = request_dict.getValue("cgac_code")
+
+        if file_type in ['D1', 'D2']:
```

Comment:
I think we should. I wanted to plan ahead for possibly generating other files outside of a submission but not sure if that's absolutely necessary at this point. What do you think?